pkg/fleetautoscalers/fleetautoscalers.go

```diff
@@ -41,94 +41,147 @@ var client = http.Client{
 
 // computeDesiredFleetSize computes the new desired size of the given fleet
 func computeDesiredFleetSize(fas *autoscalingv1.FleetAutoscaler, f *agonesv1.Fleet) (int32, bool, error) {
-
 	switch fas.Spec.Policy.Type {
 	case autoscalingv1.BufferPolicyType:
 		return applyBufferPolicy(fas.Spec.Policy.Buffer, f)
 	case autoscalingv1.WebhookPolicyType:
 		return applyWebhookPolicy(fas.Spec.Policy.Webhook, f)
 	}
 
-	return f.Status.Replicas, false, errors.New("wrong policy type, should be one of: Buffer, Webhook")
+	return 0, false, errors.New("wrong policy type, should be one of: Buffer, Webhook")
 }
 
-func applyWebhookPolicy(w *autoscalingv1.WebhookPolicy, f *agonesv1.Fleet) (int32, bool, error) {
-	faReq := autoscalingv1.FleetAutoscaleReview{
-		Request: &autoscalingv1.FleetAutoscaleRequest{
-			UID:       uuid.NewUUID(),
-			Name:      f.Name,
-			Namespace: f.Namespace,
-			Status:    f.Status,
-		},
-		Response: nil,
-	}
-	b, err := json.Marshal(faReq)
-	urlStr := ""
+func buildURLFromWebhookPolicy(w *autoscalingv1.WebhookPolicy) (*url.URL, error) {
+	var u *url.URL
+	var err error
+
 	if w.URL != nil {
-		urlStr = *w.URL
-	}
-	var faResp autoscalingv1.FleetAutoscaleReview
-	servicePath := ""
-	if w.Service != nil {
-		if w.Service.Path != nil {
-			servicePath = *w.Service.Path
+		if *w.URL == "" {
+			return nil, errors.New("URL was not provided")
 		}
+
+		u, err = url.ParseRequestURI(*w.URL)
 		if err != nil {
-			return f.Status.Replicas, false, err
+			return nil, err
+		}
+	} else {
+		if w.Service.Name == "" {
+			return nil, errors.New("service name was not provided")
+		}
+
+		var servicePath string
+		if w.Service.Path != nil {
+			servicePath = *w.Service.Path
 		}
 
 		if w.Service.Namespace == "" {
 			w.Service.Namespace = "default"
 		}
-		scheme := "http://"
+
+		scheme := "http"
 		if w.CABundle != nil {
-			scheme = "https://"
+			scheme = "https"
+
+			err = setCABundle(w.CABundle)
+			if err != nil {
+				return nil, err
+			}
 		}
-		urlStr = fmt.Sprintf("%s%s.%s.svc:8000/%s", scheme, w.Service.Name, w.Service.Namespace, servicePath)
+
+		u = createURL(scheme, w.Service.Name, w.Service.Namespace, servicePath)
```

Comment:
createURL is only used once and just sets fields on a struct, so I don't see why it isn't inlined.

Reply:
I've created a separate method in order to write several simple unit tests to check that URL corresponds to the following pattern - _name.namespace.svc:8000_
https://github.com/googleforgames/agones/pull/1531/files#diff-81c1d605645c546c9896631e7db55addR582

Reply:
That makes sense. Can you add a comment to the createURL function so that this is clear to future maintainers?

Reply:
done